docs/SIL/Instructions.md

```diff
@@ -1964,39 +1964,118 @@ destroy the value, such as `release_value`, `strong_release`,
 ### mark_dependence
 
 ```
-sil-instruction :: 'mark_dependence' '[nonescaping]'? sil-operand 'on' sil-operand
+sil-instruction :: 'mark_dependence' mark-dep-option? sil-operand 'on' sil-operand
+mark-dep-option ::= '[nonescaping]'
+mark-dep-option ::= '[unresolved]'
 
 %2 = mark_dependence %value : $*T on %base : $Builtin.NativeObject
 ```
 
 `%base` must not be identical to `%value`.
 
-Indicates that the validity of `%value` depends on the value of `%base`.
+The value of the result depends on the value of `%base`.
 Operations that would destroy `%base` must not be moved before any
-instructions which depend on the result of this instruction, exactly as
+instructions that depend on the result of this instruction, exactly as
 if the address had been directly derived from that operand (e.g. using
 `ref_element_addr`).
 
-The result is the forwarded value of `%value`. `%value` may be an
-address, but it could be an address in a non-obvious form, such as a
-Builtin.RawPointer or a struct containing the same.
-
-`%base` may have either object or address type. In the latter case, the
-dependency is on the current value stored in the address.
-
-The optional `nonescaping` attribute indicates that no value derived
-from `%value` escapes the lifetime of `%base`. As with escaping
-`mark_dependence`, all values transitively forwarded from `%value`
-must be destroyed within the lifetime of `base`. Unlike escaping
-`mark_dependence`, this must be statically verifiable. Additionally,
-unlike escaping `mark_dependence`, nonescaping `mark_dependence` may
-produce a value of non-`Escapable` type. A non-`Escapable`
-`mark_dependence` extends the lifetime of `%base` into copies of
-`%value` and values transitively forwarded from those copies. If the
-`mark_dependence` forwards an address, then it extends the lifetime
-through loads from that address. Unlike escaping `mark_dependence`, no
-value derived from `%value` may have a bitwise escape (conversion to
-UnsafePointer) or pointer escape (unknown use).
+The result is the forwarded value of `%value`. If `%value` is an
+address, then result is also an address, and the semantics are the
+same as the non-address form: the dependency is on any value derived
```

Comment:
I feel we need a definition what "derived" means.
In your example you say that a loaded value is a derived value. But what e.g. if the loaded value is stored again to another memory location. Is this "derived" as well? What about copied values? Etc.